include/oneapi/internal/math/reference_gemv.h

```diff
@@ -0,0 +1,71 @@
+/* file: reference_gemv.h */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Wrappers for BLAS functions.
+//--
+*/
+
+#ifndef __ONEAPI_INTERNAL_MATH_REFERENCE_GEMV_H__
+#define __ONEAPI_INTERNAL_MATH_REFERENCE_GEMV_H__
+
+#include "oneapi/internal/math/types.h"
+#include "oneapi/internal/types_utils.h"
+#include "services/buffer.h"
+#include "services/env_detect.h"
+
+namespace daal
+{
+namespace oneapi
+{
+namespace internal
+{
+namespace math
+{
+namespace interface1
+{
+/** @ingroup oneapi_internal
+ * @{
+ */
+
+/**
+ *  <a name="DAAL-CLASS-ONEAPI-INTERNAL__REFERENCEGEMV"></a>
+ *  \brief Adapter for reference GEMV routine
+ */
+template <typename algorithmFPType>
+struct DAAL_EXPORT ReferenceGemv
```

Comment:
You need to implement reference by based GEMM (use GEMM for GEMV for reference case).